share/itaka.1
.TH "ITAKA" "1" 
.SH "NAME" 
itaka \(em On-demand screenshooting server 
.SH "SYNOPSIS" 
.PP 
\fBitaka\fR 
.SH "DESCRIPTION" 
.PP 
Itaka is a on-demand screenshooting server based on the HTTP protocol. It integrates a specifically coded server to take screenshots on-demand when your machine is queried through a remote browser. It has a GTK GUI for client notification. 
.SH OPTIONS
Itaka follows the usual GNU command line syntax, with long
options starting with two dashes (`-').
A summary of options is included below.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-version
Show version of program.
.TP
.B \-r, \-\-revision
Show the revision of program.
.TP
.B \-d, \-\-debug
Start in debug mode.
.SH "SEE ALSO" 
.PP 
README file in source or in your distribution's documents directory.
.SH "AUTHOR" 
.PP 
Itaka was written by: Marc E. <santusmarc@users.sf.net>
.PP 
This manual page was written by Marc E. <santusmarc@users.sourceforge.net> for 
the \fBDebian\fP system (but may be used by others).  Permission is 
granted to copy, distribute and/or modify this document under 
the terms of the GNU General Public License, Version 2 any  
later version published by the Free Software Foundation. 
 
.PP 
On Debian systems, the complete text of the GNU General Public 
License can be found in /usr/share/common-licenses/GPL. 
 
.\" created by instant / docbook-to-man, Tue 05 Jun 2007, 21:05 


ChangeLog
0.3 (Two years later, I am still here):
 * Added interface translations
 * Added timestamps and enhanced error reporting to the log
 * Added a favicon to the server
 * Switch to GPLv3
 * Fixed preference pane issues for low resolution displays and some minor issues with popular GTK themes
 * Fixed duplication of Twisted reactor
 * Better console help and handling of SIGINT
 * Don't check for system compatibility of Libnotify
 * Code cleanup towards PEP-8 compliance

0.2.2:
 * Fixed a fatal bug with Python 2.6/PyGTK 2.14
 * Fixed an issue with the right images showing up in notifications
 * Fixed an fatal issue with starting Itaka from unexistant directories
 * Fixed a bug where screenshot files would be saved under the wrong
 extension/path and not erased after quitting
 * Vastly improved Makefile to support installation on different PREFIX

0.2.1:
 * Fixed a serious security bug in which /screenshot prompted no auth. Also
 fixed sending the wrong Content-Lenght for the screenshot. Report Itaka
 version on HTTP headers
 * Added RPM packages (backport from trunk) and minor UI fixes
 * Vastly improved console argument handling
 * Improved man page

0.2 (I've been watching you...):
 * Rewrote the screenshot module completely to support active window
 screenshots and scaling. #3 #10 
 * Added Basic HTTP Authentication.
 * Notifications now show up relative to the status icon.
 * Added option (-debug) to show all console output #5
 * Add options to report errors in the system tray (blinking). #7
 
 * Updated GUI on Windows not to show widgets for features that are not
 available (libnotify, screenshot active window)

 * Completely rewrote the console output and Gui logging engines. Simplified
 and fixed many isssues. Some parts of the Gui were also overhauled.
 * Automatically scroll to the last event on both logs.

 * Removed all useless type-checking and started relying on my own exceptions
 and error handlers.

 * Completely separated and encapsulated the Twisted Server code.
  
 * Improved the configuration engine to convert numbers and bools into int and
 bool rather than str. Also the Gui now saves on a per-key basis and not on 
 a all-or-nothing dict differential #9
 * Rewrote parts of the configuration engine to better handling upgrading from
 previous configuration files that are missing sections or options. #12

 * Recommented the whole code according to Epytext Markup and also did major
 tweaks to naming. #8
 * Changed strings in text to be more literal and intuitive 

 * Updated documentation and added new references.

 * Minor improvements to Debian build process.
 * Fixed minor UI quirks.
 * Fixed a syntax bug in uigtk.py when GTK version is < 2.10
 * Fixed bug when saving preferences in the GUI reverted to /tmp by default.

0.1.1:
 * Fix Debian Makefile
 * Remove bad print call in notify procedures

0.1 (Mercado):
 * Initial release


itaka.py
#! /usr/bin/env python
# -*- coding: utf8 -*-
#
# Itaka is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# any later version.
# 
# Itaka is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with Itaka; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#
# Copyright 2003-2009 Marc E.
# http://itaka.jardinpresente.com.ar
#
# $Id$ 

""" Itaka core """

import sys, traceback

validarguments = ('-h', '--help', '-v', '--version', '-r', '--revision', '-d', '--debug')
arguments = sys.argv

# Only one option at a time
if len(arguments) >= 2 and ((arguments[-1] not in validarguments) or (arguments[-1] in (validarguments[0], validarguments[1]))):
    print """Usage:
  %s [OPTION...]

  Help Options:
  -h, --help\t\t\t\tShow help options
  -v, --version\t\t\t\tShow Itaka version
  -r, --revision\t\t\tShow Itaka revision

  Application Options:
  -d, --debug\t\t\t\tStart in debug mode""" % arguments[0]
    sys.exit(1)

# Itaka core modules
try:
    # Initiate our Console and Configuration engines
    import console
    import config as itakaglobals
    configinstance = itakaglobals.ConfigParser(arguments)
    configinstance.load()

    try:
        # Initiate console with a reference to our global configuration values
        console = console.Console(itakaglobals)
    except:
        print "[*] ERROR: Could not initiate Console engine"
        traceback.print_exc()
        sys.exit(1)

    import uigtk as igui
except ImportError:
    print "[*] ERROR: Failed to import Itaka modules"
    if itakaglobals.output['debug']:
        traceback.print_exc()
    sys.exit(1)

if __name__ == "__main__":
    if len(arguments) >= 2: 
        if (arguments[-1] in (validarguments[2], validarguments[3])):
            print itakaglobals.version
            sys.exit(0)
        if (arguments[-1] in (validarguments[4], validarguments[5])):
            print itakaglobals.revision
            sys.exit(0)

    try:
        gui = igui.Gui(console, (itakaglobals, configinstance))
        gui.main()
    except Exception, e:
        console.failure(('Itaka', 'core'), "Could not initiate GUI: %s" % (e), 'ERROR')
        if itakaglobals.output['debug']:
            traceback.print_exc()
        sys.exit(1)
